// hashnode-sdk-js/dist/create-publication-article.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const base_1 = require("./base");
/**
 * Create an article in a publication.
 *
 * @param publicationId The publication's id.
 * @param article The article to be created.
 */
const createPublicationArticle = (publicationId, article) => base_1.query(`mutation createPublicationStory($input: CreateStoryInput!, $publicationId: String!) {
       createPublicationStory(input: $input, publicationId: $publicationId) {
         success,
         message,
         post {
           _id,
           title,
           slug,
           isRepublished,
           publication {
             domain
           }
         },
       }
     }`, {
    input: {
        title: article.title,
        slug: article.slug,
        tags: [],
        contentMarkdown: article.contentMarkdown,
        isRepublished: article.canonicalURL
            ? {
                originalArticleURL: article.canonicalURL,
            }
            : null,
    },
    publicationId,
}).then(({ data: { createPublicationStory } }) => {
    if (!createPublicationStory.success)
        throw new base_1.APIError([createPublicationStory.message]);
    const { post } = createPublicationStory;
    return {
        id: post._id,
        title: post.title,
        slug: post.slug,
        url: `https://${post.publication.domain}/${post.slug}`,
        canonicalURL: post.isRepublished ? article.canonicalURL : undefined,
        contentMarkdown: article.contentMarkdown,
    };
});
exports.default = createPublicationArticle;
