// hashnode-sdk-js/dist/find-user.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const base_1 = require("./base");
/**
 * Find a user by username.
 *
 * @param username The user's username.
 */
const findUser = (username) => __awaiter(void 0, void 0, void 0, function* () {
    return base_1.query(`query User($username: String!) {
       user(username: $username) {
         _id,
         username,
         name,
         tagline,
         publication {
           _id,
           title
         },
         publicationDomain
       }
     }`, { username }).then(({ data: { user } }) => ({
        id: user._id,
        username: user.username,
        name: user.name,
        tagline: user.tagline,
        publication: {
            id: user.publication._id,
            name: user.publication.title,
            domain: user.publicationDomain,
        },
    }));
});
exports.default = findUser;
